### PR TITLE
Feat simplify config

### PR DESCRIPTION
**Pull request by like-a-bause**
Starting with simplifying the Cors config on the public endpoint.

# Implementation

# Tests

# Todos

# Additional context

### CODE REVIEW

File: backend/server/public_router.go [hunk 1]
```diff
@@ -24,17 +26,29 @@ func NewPublicRouter(cfg *config.Config, persister persistence.Persister, promet
...
-			AllowOrigins:     cfg.Server.Public.Cors.AllowOrigins,
-			AllowMethods:     cfg.Server.Public.Cors.AllowMethods,
-			AllowHeaders:     cfg.Server.Public.Cors.AllowHeaders,
-			ExposeHeaders:    cfg.Server.Public.Cors.ExposeHeaders,
-			AllowCredentials: cfg.Server.Public.Cors.AllowCredentials,
-			MaxAge:           cfg.Server.Public.Cors.MaxAge,
-		}))
+	allowedOrigins := []string{}
+	isLocalhost := false
+	for _, v := range cfg.Webauthn.RelyingParty.Origins {
```
Comment (like-a-bause) [inline backend/server/public_router.go:31]: Origin could be APK Key Hash.... Or Domain without scheme://

File: backend/server/public_router.go [hunk 2]
```diff
@@ -24,17 +26,29 @@ func NewPublicRouter(cfg *config.Config, persister persistence.Persister, promet
...
+		}
+		allowedOrigins = append(allowedOrigins, v)
 	}
-	
+	e.Use(middleware.CORSWithConfig(middleware.CORSConfig{
+		AllowOrigins: allowedOrigins,
+		// only on localhost this is allowed.
+		UnsafeWildcardOriginWithAllowCredentials: isLocalhost,
+		// Needed for cross-domain
+		ExposeHeaders: []string{"X-Auth-Token"},
```
Comment (like-a-bause) [inline backend/server/public_router.go:45]: only needed when config.Session.EnableAuthTokenHeader is true

To be investigated: 
can 
```
c.Response().Header().Set("Access-Control-Expose-Headers", "X-Auth-Token")
```
 settings in handlers be deleted?

**Comment by like-a-bause** [comment]
When testing please try out in a X-Domain Setting. I pushed a docker container:
ghcr.io/teamhanko/hanko:feat-simplify-config-a820ccd

**Comment by bjoern-m** [comment]
The `backend/docs/Config.md` file needs to be updated accordingly.

File: backend/server/public_router.go [hunk 3]
```diff
@@ -24,16 +25,25 @@ func NewPublicRouter(cfg *config.Config, persister persistence.Persister, promet
...
+		httplimit.HeaderRateLimitLimit,
+		httplimit.HeaderRateLimitRemaining,
+		httplimit.HeaderRateLimitReset,
 	}
+	if cfg.Session.EnableAuthTokenHeader {
+		exposeHeader = append(exposeHeader, "X-Auth-Token")
+	}
+	e.Use(middleware.CORSWithConfig(middleware.CORSConfig{
+		// Rationale: It does not make sense to have different settings for webauthn Origin and Cors Origins.
+		// We wan't to have both in sync.
```
Comment (like-a-bause) [inline backend/server/public_router.go:39]:
```suggestion
```

File: backend/README.md [hunk 4]
```diff
@@ -290,12 +292,26 @@ Cross-Origin Resource Sharing (CORS) and specify your application(s) as allowed
...
+    cors:
+      allow_origins:
+        - "*"
+      unsafe_wildcard_origin_allowed: true
+```
+
+Wildcard `*` origins can lead to cross-site attacks and when you include a `*` wildcard origin,
+we want to make sure, that you understand what you are doing, hence this flag.
+
+> Note: In most cases, the `allow_origins` list here should contain the same entries as the `webauthn.relying_party.origins` list.
```
[the remark this answers is not in the document]
Comment (FreddyDevelop) [reply backend/README.md:313]: Well, when you have an android app, then you would add the apk key hash to the `webauthn.relying_party.origins`, but not to the `server.public.cors.allow_origins`.

**Pull request merged**